cmd/gardener-resource-manager/app/app.go

```diff
@@ -194,7 +206,22 @@ func NewResourceManagerCommand() *cobra.Command {
 		rootCAControllerOpts,
 		projectedTokenMountWebhookOpts,
 	)
+	flags := cmd.Flags()
+	opts.AddFlags(flags)
 	verflag.AddFlags(cmd.Flags())
 
 	return cmd
 }
+
+type Options struct {
+	// EnableProfiling enables profiling via web interface host:port/debug/pprof/.
+	EnableProfiling bool
+	// EnableContentionProfiling enable lock contention profiling, if profiling is enabled
+	EnableContentionProfiling bool
+}
+
+// AddFlags adds gardener-resource-manager's flags to the specified FlagSet.
+func (o *Options) AddFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&o.EnableProfiling, "profiling", true, "Enable profiling via web interface host:port/debug/pprof/")
+	fs.BoolVar(&o.EnableContentionProfiling, "contention-profiling", false, "Enable lock contention profiling, if profiling is enabled")
+}
```

Comment:
You could update `docs/monitoring/profiling.md` with instructions on how to get profiling data for GRM.